diff --git a/src/duckstation-nogui/CMakeLists.txt b/src/duckstation-nogui/CMakeLists.txt
index d362c7665..668e8a94c 100644
--- a/src/duckstation-nogui/CMakeLists.txt
+++ b/src/duckstation-nogui/CMakeLists.txt
@@ -31,6 +31,10 @@ endif()
 
 if(USE_DRMKMS)
   target_compile_definitions(duckstation-nogui PRIVATE "-DWITH_DRMKMS=1")
+  target_sources(duckstation-nogui PRIVATE
+    gbm_shim.c
+    wayland_egl_shim.c
+  )
 endif()
 
 if(WIN32)
