--- a/src/globals.h
+++ b/src/globals.h
@@ -191,6 +191,7 @@
 bool isRG353V();
 bool isRG353M();
 bool isRG552();
+bool isR50H();
 bool hasDeviceRotatedScreen();
--- a/src/globals.cpp
+++ b/src/globals.cpp
@@ -400,6 +400,10 @@
 bool isRG552()
 {
     return checkDeviceName((char *)"RG552"); 
 }
+bool isR50H()
+{
+    return checkDeviceName((char *)"R50H");
+}
 bool hasDeviceRotatedScreen(){
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -150,7 +150,7 @@
     if (!gpio_joypad)
     {
-        if (isRG351MP() || isRG552())
+        if (isRG351MP() || isRG552() || isR50H())
             gpio_joypad = true;
     }