--- a/src/globals.h
+++ b/src/globals.h
@@ -192,6 +192,7 @@
 bool isRG351V();
 bool isRG351MP();
 bool isRG552();
+bool isR50H();
 bool isRG503();
 bool isRG353V();
 bool isRG353M();
--- a/src/globals.cpp
+++ b/src/globals.cpp
@@ -404,6 +404,10 @@
 {
     return checkDeviceName((char *)"RG552"); 
 }
+bool isR50H()
+{
+    return checkDeviceName((char *)"R50H");
+}
 bool isRG503()
 {
     return checkDeviceName((char *)"RG503"); 
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -150,7 +150,7 @@
 
     if (!gpio_joypad)
     {
-        if (isRG351MP() || isRG552())
+        if (isRG351MP() || isRG552() || isR50H())
             gpio_joypad = true;
     }